ecodse_funtime_alpha/train.py

```diff
@@ -20,13 +20,14 @@ def train_loop(dataset, model, optimizer):
 
 
 def fit_loop(dataset, model, optimizer, nepoch, batchsize):
+    nstep = tf.data.experimental.cardinality(dataset).numpy() // batchsize
```

Comment:
Are you sure you want a floor division for that?
That implies that if the last batch is not a full batch it will never be seen.

Reply:
replaced with math.ceil